packages/mobile/src/hooks/useDrawer.ts

```diff
@@ -14,11 +16,32 @@ import { Drawer, setVisibility } from 'app/store/drawers/slice'
  */
 export const useDrawer = (
   drawer: Drawer
-): [boolean, (isVisible: boolean) => void] => {
+): [boolean, (isVisible: boolean) => void, boolean | 'closing'] => {
   const dispatch = useDispatch()
   const isOpen = useSelector(getVisibility(drawer))
-  const setIsOpen = (visible: boolean) =>
-    dispatch(setVisibility({ drawer, visible }))
+  const setIsOpen = (visible: boolean, closed?: boolean) => {
+    let visibleStatus = visible ? true : ('closing' as const)
+    if (closed) {
+      visibleStatus = false
+    }
+    dispatch(setVisibility({ drawer, visible: visibleStatus }))
+  }
+
+  return [isOpen === true, setIsOpen, isOpen]
```

Comment:
Can we rename `isOpen` to something like state/status/similar? this was a bit confusing to me at first

Reply:
yeah down